cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/concepts/Operation.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph.concepts
+
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.OverlayNode
+
+/**
+ * Represents an operation executed on/with a [Concept] (stored in [concept]). This is typically a
+ * `write` on a file or log object or an `execute` on a database.
+ */
+abstract class Operation : OverlayNode() {
+    /** All concept nodes are connected to an original cpg [Node] by this. */
+    abstract val cpgNode: Node
```

Comment:
I would name this node `astNode`, as it specifically connects to ast nodes and I would consider overlay nodes being cpg nodes to after they are added. I am also thinking of calling them `underlayNode` but that may be a little to weird.

Reply:
I'm not a big fan of `astNode` because we might want to connect one of the new nodes with something that's not part of the AST.
I named it `cpgNode` to point out that this is the original/classic/raw/... CPG node.
I'm happy to change this name once we have something better. `underlayingNode` would work for me.

Reply:
`underlayingNode` sounds good. And this property should be moved to the `OverlayNode`

Reply:
Done